### DISCUSSION TITLE
v4 - turbo breaks user menu on navigation

### DISCUSSION THREAD

**Discussion by thisismydesign** [activeadmin/activeadmin#8578] [Help]
Using v4.0.0.beta13.  Navigating in the menu breaks the profile menu, so that it no longer opens on clicks after navigating in the menu. This seems to happen because of turbo. Removing `import "@hotwired/turbo-rails"` from `application.js` resolves the issue.

Possible solution:
`application.js`
```js
document.addEventListener("turbo:load", () => {
  // trigger AA re-init
});
```

**Comment by tagliala** [comment]
Hello, is it possible to have a reproducible test case?

Is it related to a problem where from the main application a user follows a link to the admin area?

PS: Please also use beta15 for the reproducible test case

**Comment by thisismydesign** [comment]
I have an open source app that reproduces it: https://github.com/thisismydesign/wealth/pull/6/commits/0c96b799f13ed4017788031257cec514dc4a6961

`docker-compose up` should work out of the box with credentials in `db/seeds/development.rb`.

Otherwise, it's a simple rails 7 + importmaps + aa v4 setup.

Is there something I could call in the turbo hook to re-initialize AA's JS?

**Comment by tagliala** [comment]
Thanks, I've tried to run it in docker but assets don't work (macOS Sequoia 15.2, Apple Silicon, docker 4.37.0 without login)

I can see that activeadmin is using a custom head partial which is importing application.js, but I don't know if this is a good practice

I would keep them separated.

> Is there something I could call in the turbo hook to re-initialize AA's JS?

Maybe it is flowbite https://github.com/activeadmin/activeadmin/discussions/8573

If it is flowbite support, please close this issue so we can use #8573 for discussion

**Reply by thisismydesign** [reply]
Yes, this is the same issue
